lib/orbit-common/transaction.js

```diff
@@ -0,0 +1,111 @@
+import Orbit from 'orbit/main';
+import MemorySource from './memory-source';
+import { assert } from 'orbit/lib/assert';
+import { coalesceOperations } from 'orbit/lib/operations';
+
+/**
+ `Transaction` is a form of `MemorySource` that isolates changes from a base
+ source.
+
+ A transaction has its own internal cache. By default, this cache will
+ `retrieve` data from the base source if it's not defined. Alternatively, a
+ transaction can be completely isolated from the base source by setting the
+ `isolated` option to `true`. An isolated transaction's cache must be
+ manually managed.
+
+ Transactions are "begun" immediately by default. This means that they
+ immediately begin tracking transforms applied to them. When `commit` is
+ invoked, the transforms applied to the transaction will be coalesced and
+ applied to the base source.
+
+ @class Transaction
+ @extends MemorySource
+ @namespace OC
+ @param {OC.Schema} schema
+ @param {Object}    [options]
+ @param {OC.Source} [options.baseSource] The transaction's base source (required).
+ @param {Boolean}   [options.active=true] Should this transaction begin immediately on creation?
+ @param {Boolean}   [options.isolated=false] Should this transaction's cache be completely isolated from the base source?
+ @constructor
+ */
+var Transaction = MemorySource.extend({
+  active: false,
+  isolated: false,
+  operations: null,
+  inverseOperations: null,
+
+  init: function(schema, options) {
```

Comment:
`schema` is unused here -- should we

```js
assert('`schema` if provided must equal `options.baseSource.schema`', !schema || schema === options.baseSource.schema);
```

or possibly add the logic to not pass `schema` at all and detect:
```js
if (schema.baseSource) {
  options = schema;
}